File: JavaScript/8js 4/stringClean.js
/*
|--------------------------------------------------------------------------
| String cleaning
|--------------------------------------------------------------------------
|
*/

function stringClean(s) {
   return [...s].filter((i) => !"0123456789".includes(i)).join("");
}


function stringClean(s) {
   return s.replace(/\d/g, "");
}


function stringClean(s) {
   return s.replace(/[0-9]/g, "");
}


console.log(stringClean("! !"), "! !");
console.log(stringClean("123456789"), "");
console.log(stringClean("(E3at m2e2!!)"), "(Eat me!!)");

File: Python/8py 4/string_clean.py
"""
|--------------------------------------------------------------------------
| String cleaning
|--------------------------------------------------------------------------
|
"""

def string_clean(s):
   return ''.join([i for i in s if i not in "0123456789"])


def string_clean(s):
   for d in '0123456789':
      s = s.replace(d, '')
   return s


def string_clean(s):
   return ''.join(x for x in s if not x.isdigit())


def string_clean(s):
   return s.translate(s.maketrans(' ', ' ', '1234567890'))


print(string_clean("! !"), "! !")
print(string_clean("123456789"), "")
print(string_clean("(E3at m2e2!!)"), "(Eat me!!)")